### PR TITLE
Re-enable a few tests

### LINKED ISSUE #1077: Update VS install on CI machines to a version with the bundled .NET Core / .NET Standard .props, and re-enable tests
In #1052, I'm removing the `RuntimeFrameworkVersion` from test projects that target .NET Core 2.0.  This doesn't work with the version of VS installed on the CI machines yet, so I'm disabling the following tests on full framework:

- Microsoft.NET.Build.Tests.GivenThatWeWantToReferenceAnAssembly.ItRunsAppsDirectlyReferencingAssemblies
- Microsoft.NET.Build.Tests.GivenThatWeWantToVerifyProjectReferenceCompat.Project_reference_compat (when targeting .NET Core 2.0)
- Microsoft.NET.Build.Tests.GivenThatWeWantToVerifyNuGetReferenceCompat.Nuget_reference_compat (when targeting .NET Core 2.0 or .NET Standard 2.0)
- Microsoft.NET.Publish.Tests.GivenThatWeWantToPublishAHelloWorldProject.Conflicts_are_resolved_when_publishing_a_portable_app
- Microsoft.NET.Publish.Tests.GivenThatWeWantToPublishAHelloWorldProject.Conflicts_are_resolved_when_publishing_a_self_contained_app
- Microsoft.NET.Publish.Tests.GivenThatWeWantToPublishAHelloWorldProject.Publish_standalone_post_netcoreapp2_app_and_it_should_run
- Microsoft.NET.Publish.Tests.GivenThatWeWantToPublishAHelloWorldProject.Conflicts_are_resolved_when_publishing_a_portable_app

We need to update the CI machines to a newer version of VS and then re-enable these tests.

See also #1041, we will probably fix both of these together.

### LINKED ISSUE #1293: Tool deps file generation tests fail when run on full Framework MSBuild
The tests in `GivenThatWeWantToGenerateADepsFileForATool` are failing when running on full Framework MSBuild with the following error:

> C:\git\dotnet-sdk\bin\Debug\Sdks\Microsoft.NET.Sdk\build\GenerateDeps\GenerateDeps.proj(29,5): error MSB4184: The expression "[System.IO.Path]::GetDirectoryName("C:\git\dotnet-sdk\packages\.tools\testtool\1.0.0\netcoreapp2.0\project.assets.json")" cannot be evaluated. Illegal characters in path.

The scenarios these tests cover aren't applicable to full framework MSBuild, but it is unexpected that they would fail in this way.  We should investigate what is going on.

### PR DESCRIPTION
**Pull request by dsplaisted**
Fixes #1077
Fixes #1293

### CODE REVIEW

**Comment by dsplaisted** [comment]
@dotnet-bot test OSX10.12 Debug

**Review by livarcocc — APPROVED** [review]
Once CI is green.

**Comment by dsplaisted** [comment]
@dotnet-bot test OSX10.12 Release

**Comment by dsplaisted** [comment]
@dotnet-bot test OSX10.12 Debug

**Comment by dsplaisted** [comment]
@dotnet-bot test OSX10.12 Debug
test Ubuntu14.04 Debug

**Comment by dsplaisted** [comment]
@livarcocc @nguerrera @wli3 @tannergooding @johnbeisner @peterhuene This is ready for review now.  It's changed a bunch since Livar originally reviewed it.

File: src/Assets/NuGet.Config [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<configuration>
+  <packageSources>
+    <add key="dotnet-core" value="https://dotnetfeed.blob.core.windows.net/dotnet-core/packages/index.json" />
```
Comment (livarcocc) [inline src/Assets/NuGet.Config]: My only concern is that this won't break prod-con. @johnbeisner or @tannergooding please speak up to it.

Comment (dsplaisted) [reply src/Assets/NuGet.Config]: I chatted with Tanner about this and it sounds like this is the feed that will be used for prod-con.  There is some capability to inject additional feeds in the build infrastructure, but it's not being used and wouldn't have flowed through to the tests before this PR either.

**Pull request merged**